lib/src/mock_storage_reference.dart

```diff
@@ -77,4 +82,46 @@ class MockReference extends Mock implements Reference {
   Future<Uint8List> getData([int maxSize = 10485760]) {
     return Future.value(_storage.storedDataMap[_path]);
   }
+
+  @override
+  Future<FullMetadata> updateMetadata(SettableMetadata metadata) {
+    // ignore: omit_local_variable_types
+    final newSettable = _getNewSettableFromFullMetadata(_storage.storedMetadata[_path], metadata);
+    _storage.storedMetadata[_path] = _createFullMetadata(newSettable);
+    return Future<FullMetadata>.value(_storage.storedMetadata[_path]);
+  }
+
+  FullMetadata _createFullMetadata(SettableMetadata? metadata) {
```

Comment:
Okay, thank you for your hints, they were really useful!

Just one thing

> Btw, since the `bucket`, `fullPath`, etc. properties aren't ever modifiable, it'd be more readable to generate the values everytime they are needed, instead of generating them the first time you run `_createFullMetadata`, and storing them into `storedMetadata`. Something like:
> 
> ```dart
>   Map<String, dynamic> getGeneratedMetadata() {
>     return {
>       'bucket': bucket,
>       'fullPath': fullPath,
>       ...
>     };
>   }
> ```

I agree with you on what you've said except for the two fields `creationTimeMillis` and `updateTimeMillis`.. we should store them when a file is uploaded and then we should be able to check the original value later... I don't know if I explained myself